pgbackrest: 2.55.1 -> 2.56.0

iedame opened the pull request:
fixes https://github.com/NixOS/nixpkgs/pull/427240
changelog: https://github.com/pgbackrest/pgbackrest/releases/tag/release%2F2.56.0

wip
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [x] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

iedame committed:
pgbackrest: 2.55.1 -> 2.56.0

iedame commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 443717`
Commit: [`5cf8832e80b39e788f1c66375d767b32981fe4cf`](https://github.com/NixOS/nixpkgs/commit/5cf8832e80b39e788f1c66375d767b32981fe4cf) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/5cf8832e80b39e788f1c66375d767b32981fe4cf..pull/443717/head))
Merge: [`fc7ad135c9d2c672980215816a2ce87ac53262b5`](https://github.com/NixOS/nixpkgs/commit/fc7ad135c9d2c672980215816a2ce87ac53262b5)

Logs: https://github.com/iedame/nixpkgs-review-gha/actions/runs/17797208543


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>pgbackrest</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>pgbackrest</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>pgbackrest</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>pgbackrest</li>
  </ul>
</details>

iedame commented on nixos/tests/pgbackrest/sftp.nix:
```diff
@@ -77,6 +79,11 @@ in
       """))
 
       with subtest("backup/restore works with local instance/remote repo (SFTP)"):
+        # Create and set permissions for the lock file directory
+        primary.succeed("mkdir -p /tmp/pgbackrest")
+        primary.succeed("chown pgbackrest:pgbackrest /tmp/pgbackrest")
+        primary.succeed("chmod 770 /tmp/pgbackrest")
+
```
I never use pgbackrest with SFTP so I will do some digging.

wolfgangwalther reviewed (commented):
This commit seems to have added a requirement for a lock to `restore`: https://github.com/pgbackrest/pgbackrest/commit/8cdd9ce1c4ab6cca508932a41a3013374d7547ef#diff-fca94c6118f5942f673a94140c6f9134db3413250a31308848c578ca7f12eb1c

The docs tell me that there is a `--lock-path` option that could potentially be helpful: https://pgbackrest.org/command.html

It's interesting that only the `sftp` test fails, but not the `posix` test. The commit mentioned above also mentions something like:

> Therefore, no lock is needed for restores with a remote role [...]

(continuing my investigation)

iedame commented on nixos/tests/pgbackrest/sftp.nix:
```diff
@@ -16,6 +16,8 @@ in
       ...
     }:
     {
+      users.users.postgres.extraGroups = [ "pgbackrest" ];
```
Taking a deep look at the module now, I probably should have done that before. Thank you for pointing that out.

wolfgangwalther committed:
nixos/pgbackrest: add commands option for specific settings

These settings are documented here:
https://pgbackrest.org/configuration.html

It's useful to be able to define specific configuration for separate
commands, for example for fixing a failing test in the next update.

iedame commented:
Thank you for your patience and for sharing your thought process. I'm learning a lot from this.

I'm still running into a permission error when trying to run the tests. I'm not sure if this is an issue on my end or if it's expected behavior after the changes.

iedame commented:
That makes sense!

I can confirm that after changing the path to `/tmp/postgresql`, the tests now pass successfully.

I have pushed the change and will now proceed with running the remaining tests and `nixpkgs-review`.
